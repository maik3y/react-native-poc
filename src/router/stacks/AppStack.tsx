import { createBottomTabNavigator } from '@react-navigation/bottom-tabs'
import { NavigatorScreenParams } from '@react-navigation/native'
import React from 'react'
import { useColorScheme } from 'react-native'
import { Colors } from 'react-native/Libraries/NewAppScreen'
import { ContractProps } from '../../screens/ContractsScreen/hooks/useContracts'
import ContractsIcon from '../../screens/ContractsScreen/icons/ContractenIcon'
import ScanIcon from '../../screens/ScanScreen/icons/ScanIcon'
import ScanScreen from '../../screens/ScanScreen/ScanScreen'
import TicketsIcon from '../../screens/TicketsScreen/icons/TicketsIcon'
import TransportIcon from '../../screens/TransportScreen/icons/TransportIcon'
import TransportScreen from '../../screens/TransportScreen/TransportScreen'
import ContractsScreenStack, { ContractsStackParams } from './ContractenStack'
import TicketsScreenStack, { TicketsStackParams } from './TicketsStack'

export type AppStackParams = {
  ContractsStack: NavigatorScreenParams<ContractsStackParams>
  Contract: {
    data: ContractProps
  }
  TicketsStack: NavigatorScreenParams<TicketsStackParams>
  Ticket: {
    id: number
  }
  Transport: undefined
  Scan: undefined
}

const RootStack = createBottomTabNavigator<AppStackParams>()

export default function AppStack() {
  const colorScheme = useColorScheme()

  return (
    <RootStack.Navigator
      initialRouteName="ContractsStack"
      screenOptions={() => ({
        tabBarActiveTintColor: colorScheme
          ? Colors[colorScheme].tabBarActiveTintColor
          : '',
        tabBarInactiveTintColor: colorScheme
          ? Colors[colorScheme].tabBarInactiveTintColor
          : ''
      })}>
      <RootStack.Screen
        name="ContractsStack"
        component={ContractsScreenStack}
        options={{
          title: 'Contracts',
          tabBarIcon: ({ color, size }) => (
            <ContractsIcon color={color} size={size} />
          ),
          tabBarLabel: 'Contracts'
        }}
      />
      <RootStack.Screen
        name="TicketsStack"
        component={TicketsScreenStack}
        options={{
          title: 'Tickets',
          tabBarIcon: ({ color, size }) => (
            <TicketsIcon color={color} size={size} />
          ),
          tabBarLabel: 'Tickets'
        }}
      />
      <RootStack.Screen
        name="Transport"
        component={TransportScreen}
        options={{
          tabBarIcon: ({ color, size }) => (
            <TransportIcon color={color} size={size} />
          ),
          tabBarLabel: 'Transport'
        }}
      />
      <RootStack.Screen
        name="Scan"
        component={ScanScreen}
        options={{
          tabBarIcon: ({ color, size }) => (
            <ScanIcon color={color} size={size} />
          ),
          tabBarLabel: 'Scan'
        }}
      />
    </RootStack.Navigator>
  )
}
