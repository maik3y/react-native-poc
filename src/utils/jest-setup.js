// To supress warning message in console when running tests
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')
