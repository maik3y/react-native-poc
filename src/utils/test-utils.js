import { render } from '@testing-library/react-native'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import AuthProvider from '../contexts/AuthProvider'

const queryClient = new QueryClient()

const AllTheProviders = ({ children }) => {
  return (
    <QueryClientProvider client={queryClient}>
      <AuthProvider>{children}</AuthProvider>
    </QueryClientProvider>
  )
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react-native'

// override render method
export { customRender as render }
