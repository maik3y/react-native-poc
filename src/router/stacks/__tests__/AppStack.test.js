import * as React from 'react'
import { render, screen } from '../../../utils/test-utils'

import { NavigationContainer } from '@react-navigation/native'
import AppStack from '../AppStack'

describe('AppStack', () => {
  test('Should render', () => {
    render(
      <NavigationContainer>
        <AppStack />
      </NavigationContainer>
    )
    expect(screen.toJSON()).toMatchSnapshot()
  })
})
