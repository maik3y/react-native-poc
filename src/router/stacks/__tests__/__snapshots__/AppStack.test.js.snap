// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppStack Should render 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenNavigationContainer
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <RNSScreen
      activityState={2}
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 0,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          Array [
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
              "flex": 1,
            },
            Array [
              Object {
                "flex": 1,
                "flexDirection": "column-reverse",
              },
              undefined,
            ],
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                },
                undefined,
              ]
            }
          >
            <RNSScreenStack
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <RNSScreen
                collapsable={false}
                forwardedRef={[Function]}
                gestureResponseDistance={
                  Object {
                    "bottom": -1,
                    "end": -1,
                    "start": -1,
                    "top": -1,
                  }
                }
                nativeBackButtonDismissalEnabled={false}
                onAppear={[Function]}
                onDisappear={[Function]}
                onDismissed={[Function]}
                onHeaderBackButtonClicked={[Function]}
                onNativeDismissCancelled={[Function]}
                onTransitionProgress={[Function]}
                onWillDisappear={[Function]}
                replaceAnimation="push"
                stackPresentation="push"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
                swipeDirection="horizontal"
              >
                <RNSScreenStackHeaderConfig
                  backButtonInCustomView={false}
                  backgroundColor="rgb(255, 255, 255)"
                  color="rgb(0, 122, 255)"
                  direction="ltr"
                  disableBackButtonMenu={false}
                  hidden={true}
                  hideBackButton={false}
                  largeTitleHideShadow={false}
                  title="Contracts"
                  titleColor="rgb(28, 28, 30)"
                  topInsetEnabled={false}
                  translucent={false}
                />
                <View
                  accessibilityElementsHidden={false}
                  importantForAccessibility="auto"
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "flex": 1,
                        },
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Array [
                            Object {
                              "display": "flex",
                            },
                            Object {
                              "justifyContent": "center",
                            },
                            Object {
                              "alignItems": "center",
                            },
                          ],
                        ]
                      }
                    >
                      <Text>
                        Fetching...
                      </Text>
                    </View>
                  </View>
                </View>
              </RNSScreen>
            </RNSScreenStack>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenNavigationContainer>
  <View
    collapsable={false}
    onLayout={[Function]}
    pointerEvents="auto"
    style={
      Object {
        "backgroundColor": "rgb(255, 255, 255)",
        "borderTopColor": "rgb(216, 216, 216)",
        "borderTopWidth": 0.5,
        "bottom": 0,
        "elevation": 8,
        "height": 49,
        "left": 0,
        "paddingBottom": 0,
        "paddingHorizontal": 0,
        "position": null,
        "right": 0,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <View
      accessibilityRole="tablist"
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityLabel="ContractsStack, tab, 1 of 4"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "selected": true,
          }
        }
        accessibilityStates={
          Array [
            "selected",
          ]
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <Text />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 0,
                },
              ]
            }
          >
            <Text />
          </View>
        </View>
      </View>
      <View
        accessibilityLabel="TicketsStack, tab, 2 of 4"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "selected": false,
          }
        }
        accessibilityStates={Array []}
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 0,
                },
              ]
            }
          >
            <Text />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <Text />
          </View>
        </View>
      </View>
      <View
        accessibilityLabel="Transport, tab, 3 of 4"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "selected": false,
          }
        }
        accessibilityStates={Array []}
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 0,
                },
              ]
            }
          >
            <Text />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <Text />
          </View>
        </View>
      </View>
      <View
        accessibilityLabel="Scan, tab, 4 of 4"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "selected": false,
          }
        }
        accessibilityStates={Array []}
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 0,
                },
              ]
            }
          >
            <Text />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
